### DISCUSSION TITLE
Welcome to GoPro Labs Discussions!

### DISCUSSION THREAD

**Discussion by dnewman-gpsw** [gopro/labs#85] [Announcements]
## 👋 Welcome!
  GoPro Labs very much encourages user feedback, and users helping other will Labs enabled projects. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together.

**Comment by TimRenfroe13** [comment]
I've searched the discussions for any scripting reference material but haven't had any luck.  Can / will someone point me in the proper direction?  I have seen reference to scripting on drones, do they share the same scripting engine / processor?

**Reply by dnewman-gpsw** [reply]
Labs is about controlling and scripting GoPro camera's. Nothing to do with drones directly, certainly no scripting language is shared with the drone. Every qr code you see here are mini scripts. There is documentation starting at the top https://gopro.github.io/labs/

**Comment by TimRenfroe13** [comment]
Thank you for the quick response.  I am looking for a syntax guide and any reference material to keywords, operators, special characters (!, #) purposes, etc.  Thank you again!

**Reply by dnewman-gpsw** [reply]
This information is linked above under "Primer on Labs actions, scripting and command sets".  PS # is not used.

**Comment by TimRenfroe13** [comment]
Great information, I did see this page on a tutorial page and was curious if it’s been moved somewhere that you know of….

The url is https://gopro.github.io/labs/control/set/ but the page no longer has a Manual Command text box

Here’s the page:
￼

**Reply by dnewman-gpsw** [reply]
It doesn't need one.  All QR code generators work, it is just text.  This page is for URL based qr codes. 

e.g. https://gopro.github.io/labs/control/set/?cmd=mVr4p60fW

**Comment by tchriss1** [comment]
I have an industrial related application which would require the following:

1. Acquisition of a 1 or 2 minute video (max length) at 240 fps ideally when:

   a. A Windows application which we write sends a command to the GoPro camera, or closes a mechanical switch.

2. Automatic transfer of the acquired video file to the PC without user intervention.

3.  We would use it as follows:
   
    a. When a short experiment is ready to be done, we would do whatever else necessary (ideally by software) to set up the GoPro camera for the upcoming experiment

    b. And at the appropriate time issue whatever command, or do what we need to do to start the video capture.

    c. Perhaps a few minutes after that "first experiment", we would do the same thing, and repeat perhaps up to 5 times.

    d. Then at some appropriate time, review the videos.

Is this kind of thing possible with GoPro Labs or other tools you have?   If not, can something similar be accomplished in some way?

(Potentially we would implement up to 20 or more systems, each with a camera, if this works.)


Thanks in advance.

**Reply by dnewman-gpsw** [reply]
Not for Labs, more for Open GoPro which supports camera control and data transfers over USB.  Info at https://gopro.github.io/OpenGoPro/

**Reply by tchriss1** [reply]
Thanks for the quick response.  This was very helpful

**Comment by tchriss1** [comment]
Hi David,

  Thanks so much.  I'll look things over.  If I have questions, are you the
person I should interact with or is there someone else.

Best regards,

Terry

On Wed, Sep 13, 2023 at 3:06 PM David Newman ***@***.***>
wrote:

> Not for Labs, more for Open GoPro which supports camera control and data
> transfers over USB. Info at https://gopro.github.io/OpenGoPro/
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/gopro/labs/discussions/85#discussioncomment-6995352>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BCR6LXAD2RHFTJJNJYLC6ILX2IU7LANCNFSM6AAAAAAQNW3WXU>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Terry Chriss, President
Custom Lab Software Systems, Inc.
90 Main Street, Unit 108B
Centerbrook, CT  06409
(860) 767-1645p
(860)767-1592f
www.CustomLabSoftware.com <http://www.customlabsoftware.com>

**Comment by smeyernyc** [comment]
Is there a way to lock out scanning to prevent malicious QR codes?
I typically run my GP using a remote while mounted to a vehicle. What's to stop the camera from scanning a QR code someone printed out to be funny and stuck to their rear windshield? Between remote use, the camera is actively scanning.
Also secondary question is there a way to have all other four CC codes but ownr hidden during startup?

**Reply by dnewman-gpsw** [reply]
Malicious QR code aren't very practical with a wide angle lens, as the code have to be huge or very close to the camera.  And most dangerous commands, pairing, Wi-Fi credentials and formatting, require on camera user acknowledgments. 

On HERO11 and 12 there is the new !MFAST=1 command which suppresses some of the boot messages.

**Reply by smeyernyc** [reply]
@dnewman-gpsw fair enough I haven't even tried it. I was just curious if a malicious actor can do it, and if it was avoidable. So I tried the M fast but it also sped up the owner message. Regardless of the fact that I have it set to 7000 milliseconds

**Reply by dnewman-gpsw** [reply]
Yes, FAST speeds everything as it is used make camera scripts launch faster. Another trick is use additional owner info, using OWNS and OWNT etc.

**Reply by SiliconFiend** [reply]
> Yes, FAST speeds everything as it is used make camera scripts launch faster. Another trick is use additional owner info, using OWNS and OWNT etc.

I also would like to have a fast startup with the exception of owner information. Can you explain more about OWNS and OWNT? I couldn't find where they are documented. I think I tried !MFAST=1 (Hero 11) and I believe it skipped all the labs messages including owner. Can't seem to find the documentation for FAST either (are there settings other than 0 and 1?).

**Reply by dnewman-gpsw** [reply]
Try them. FAST doesn't bypass the OWNR, just doubles it speed.  

Try

 !MOWNS="message 1"
 !MOWNT="message 2"
 !MOWNU="message 3"

1 is active 0 is deactived.

**Reply by SiliconFiend** [reply]
Looks like it's basically multiple "pages" of owner (or whatever) information? So for extended time just duplicate the information onto subsequent letters (S, T, U, etc.)?

**Comment by smeyernyc** [comment]
I know you can trigger recording by USB power, but can you stop it than gracefully shutdown? Assuming battery was installed. Use case is hardwiring "record"  switch that triggers USB power on/off.

**Reply by dnewman-gpsw** [reply]
Yes. !O <- is shutdown.  You can use a USB power conditional to call that.

**Reply by smeyernyc** [reply]
@dnewman-gpsw so !uN!S!uE!R!0 ??

**Reply by smeyernyc** [reply]
Or !uN!S!uE!0!R

**Reply by smeyernyc** [reply]
Or !mwake=2!S!uE!0!R

**Reply by dnewman-gpsw** [reply]
This is the very old way, but not so reliable, no I was recommending the new script commands.   https://gopro.github.io/labs/control/actions/

**Reply by dnewman-gpsw** [reply]
Definitely not !m.... (not a command)

!MWAKE=2 <- wake on power feature

You would need a BOOT command, and an action script based like !S<u0!E+!O

This not an appropriate thread "Welcome to GoPro Labs Discussions!"

Do some more research, do some testing, then post your questions in a new thread related to the questions.
